internal/handshake/crypto_setup_client.go

```diff
@@ -49,10 +49,11 @@ type cryptoSetupClient struct {
 	nullAEAD             crypto.AEAD
 	secureAEAD           crypto.AEAD
 	forwardSecureAEAD    crypto.AEAD
-	aeadChanged          chan<- protocol.EncryptionLevel
 
-	requestConnIDOmission bool
-	params                *paramsNegotiatorGQUIC
+	paramsReceived chan<- *TransportParameters
```

Comment:
Nit: For small structs I would say it's preferable to send a value instead of a pointer, as that makes it explicit that the value is immutable and no synchronization is needed.

Reply:
Second nit: I don't like the name of the chan, it sounds like one of the close-only chans. Why not just call `transportParams` or `paramsChan`?